### DISCUSSION TITLE
What the heck is wp-env?

### DISCUSSION THREAD

**Discussion by mburridge** [WordPress/developer-blog-content#33] [Topic Ideas]
I propose an article on local WordPress environments for developers with an emphasis on `wp-env`, explaining what it is, what the benefits of using it are, and how to use it. The post should include plenty of screenshots of `wp-env` in action.

## Outline

**Local WordPress environments**
Local WP environment - what and why
A brief survey of:
- MAMP/WAMP/XAMPP
- Local (Flywheel)
- Valet (Laravel)
- DevKinsta (Kinsta)

**What is `wp-env`?**
Allows you to spin up a transitory WordPress environment from the command line. 
Primarily for development.
Requires Docker.
Requires Node/NPM.

**Why use `wp-env`?**
Quick and easy way to spin up a temporary WordPress environment for testing and development.
Doesn't require a monolithic installation.
Zero config.
Provides a testing environment on port 8889

**Installation**
Via NPM, either globally with -g switch, or in an existing project with a `package.json` file as a dev dependency 
Test with `wp-env --version`

**Usage**
`wp-env start` from project folder, or `npm rum wp-env start` if installed as a dev dependency
If Docker is not running you'll get an error
Can run in browser with `localhost:8888` - login credentials, etc...
Stop the environment with `wp-env stop`
Can run WP-CLI commands with `wp-env run cli [command]`
Can customise environment with a `.wp-env.json` file

**Troubleshooting**
Detail various troubleshooting steps

**References**

- https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/
- https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/#wp-env-json

**Comment by bph** [comment]
Discussion is now locked. Please add further comments to the issue: 

- #89

**Comment by bph** [comment]
Published: [Quick and easy local WordPress development with wp-env](https://developer.wordpress.org/news/2023/03/quick-and-easy-local-wordpress-development-with-wp-env/)
